### PR TITLE
Add get available opsets

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Introduce new API in order to reduce the number of places with registration opsets

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/core/src/opsets/opset.cpp [hunk 1]
```diff
@@ -48,6 +48,26 @@ ov::Node* ov::OpSet::create_insensitive(const std::string& name) const {
     return m_factory_registry.create(type_info_it->second);
 }
 
+#define _OPENVINO_REG_OPSET(OPSET) opset_map[#OPSET] = ov::get_##OPSET;
+
+const std::map<std::string, std::function<const ov::OpSet&()>>& ov::get_avalable_opsets() {
```
Comment (ilya-lavrenov) [inline src/core/src/opsets/opset.cpp:53]: will it affect CC? 
here we introduce one more points which holds other code

Comment (ilyachur) [reply src/core/src/opsets/opset.cpp:53]: No it shouldn't affect. Here we hold only constrctors of opsets. But CC works with insert operations

File: src/bindings/python/src/compatibility/pyngraph/node_factory.cpp [hunk 2]
```diff
@@ -74,18 +75,7 @@ class NodeFactory {
...
-            {"opset3", OpsetFunction(ngraph::get_opset3)},
-            {"opset4", OpsetFunction(ngraph::get_opset4)},
-            {"opset5", OpsetFunction(ngraph::get_opset5)},
-            {"opset6", OpsetFunction(ngraph::get_opset6)},
-            {"opset7", OpsetFunction(ngraph::get_opset7)},
-            {"opset8", OpsetFunction(ngraph::get_opset8)},
-            {"opset9", OpsetFunction(ngraph::get_opset9)},
-            {"opset10", OpsetFunction(ngraph::get_opset10)},
-        };
+        const auto& s_opsets = ngraph::get_avalable_opsets();
```
Comment (t-jankowski) [inline src/bindings/python/src/compatibility/pyngraph/node_factory.cpp:78]:
```suggestion
        const auto& s_opsets = ngraph::get_available_opsets();
```

**Review by t-jankowski — CHANGES_REQUESTED** [review]

File: src/tests/ie_test_utils/functional_test_utils/src/summary/op_summary.cpp [hunk 3]
```diff
@@ -27,6 +27,7 @@ void OpSummaryDestroyer::initialize(OpSummary *p) {
 
 OpSummary::OpSummary() {
     reportFilename = CommonTestUtils::OP_REPORT_FILENAME;
+    // TODO: replace to get_available_opsets()
```
Comment (ilyachur) [inline src/tests/ie_test_utils/functional_test_utils/src/summary/op_summary.cpp:30]: @iefode Can we use new API?

File: src/core/src/opsets/opset.cpp [hunk 4]
```diff
@@ -207,3 +227,23 @@ const ngraph::OpSet& ngraph::get_opset10() {
...
     return opset;
 }
+
+#define _NGRAPH_REG_OPSET(OPSET) opset_map[#OPSET] = ngraph::get_##OPSET;
+
+const std::map<std::string, std::function<const ngraph::OpSet&()>>& ngraph::get_available_opsets() {
+    static std::map<std::string, std::function<const ngraph::OpSet&()>> opset_map;
+    static std::once_flag flag;
+    std::call_once(flag, [&]() {
+        _NGRAPH_REG_OPSET(opset1);
```
Comment (ilya-lavrenov) [inline src/core/src/opsets/opset.cpp:237]: 1. Why do we need ngraph and ov function? can we have a single one?
2. Why do we need once_flag? can we just create map as const object and perform assignment during construction?
3. Can we add its usage in ie_core.cpp as well?

Comment (ilyachur) [reply src/core/src/opsets/opset.cpp]: We have different opset objects for `ngraph` and `ov`.
All comments fixed.

File: src/core/src/opsets/opset.cpp [hunk 5]
```diff
@@ -48,6 +48,40 @@ ov::Node* ov::OpSet::create_insensitive(const std::string& name) const {
     return m_factory_registry.create(type_info_it->second);
 }
 
+#define _NGRAPH_REG_OPSET(OPSET) \
```
Comment (tomdol) [inline src/core/src/opsets/opset.cpp:51]: https://en.cppreference.com/w/cpp/language/identifiers

> the identifiers with a double underscore anywhere are reserved;
> the identifiers that begin with an underscore followed by an uppercase letter are reserved;
> the identifiers that begin with an underscore are reserved in the global namespace.

File: src/core/src/opsets/opset.cpp [hunk 6]
```diff
@@ -48,6 +48,40 @@ ov::Node* ov::OpSet::create_insensitive(const std::string& name) const {
...
+                                                                                           _NGRAPH_REG_OPSET(opset9),
+                                                                                           _NGRAPH_REG_OPSET(opset10)};
+#undef _NGRAPH_REG_OPSET
+    return opset_map;
+}
+
+const std::map<std::string, std::function<const ov::OpSet&()>>& ov::get_available_opsets() {
+#define _OPENVINO_REG_OPSET(OPSET) \
+    { #OPSET, ov::get_##OPSET }
+    static std::map<std::string, std::function<const ov::OpSet&()>> opset_map = {_OPENVINO_REG_OPSET(opset1),
```
Comment (ilya-lavrenov) [inline src/core/src/opsets/opset.cpp:71]: `const` here as well?

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**